Tests/Rotator/RotatorTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: jjose00
+ * Date: 11/10/17
+ * Time: 9:37 AM
+ */
+
+namespace Giftcards\Encryption\Tests\Rotator;
+
+use Giftcards\Encryption\CipherText\Rotator\Bounds;
+use Giftcards\Encryption\CipherText\Rotator\NullObserver;
+use Giftcards\Encryption\CipherText\Rotator\Record;
+use Giftcards\Encryption\CipherText\Rotator\Rotator;
+use Giftcards\Encryption\CipherText\Rotator\StoreInterface;
+use Giftcards\Encryption\CipherText\Rotator\StoreRegistry;
+use Giftcards\Encryption\Encryptor;
+use Giftcards\Encryption\Tests\AbstractTestCase;
+use Mockery;
+
+class RotatorTest extends AbstractTestCase
+{
+    public function testRotation()
+    {
+        $records = [
+            new Record(0, ['record0_encrypted']),
+            new Record(1, ['record1_encrypted']),
+            new Record(2, ['record2_encrypted']),
+            new Record(3, ['record3_encrypted']),
+        ];
+
+        $observer = new NullObserver();
+
+        $encryptor = Mockery::mock(Encryptor::class);
+        $encryptor->shouldReceive("decrypt")->with("record0_encrypted")->andReturn("record0_decrypted");
+        $encryptor->shouldReceive("decrypt")->with("record1_encrypted")->andReturn("record1_decrypted");
+        $encryptor->shouldReceive("decrypt")->with("record2_encrypted")->andReturn("record2_decrypted");
+        $encryptor->shouldReceive("decrypt")->with("record3_encrypted")->andReturn("record3_decrypted");
+
+        $encryptor->shouldReceive("encrypt")->with("record0_decrypted", "test_profile")->andReturn("record0_rotated");
+        $encryptor->shouldReceive("encrypt")->with("record1_decrypted", "test_profile")->andReturn("record1_rotated");
+        $encryptor->shouldReceive("encrypt")->with("record2_decrypted", "test_profile")->andReturn("record2_rotated");
+        $encryptor->shouldReceive("encrypt")->with("record3_decrypted", "test_profile")->andReturn("record3_rotated");
+
+        $store = Mockery::mock(StoreInterface::class);
+        $store->shouldReceive("fetch")->withArgs([0, 3])->andReturn(
+            [
+                $records[0],
+                $records[1],
+                $records[2],
+            ]);
+        $store->shouldReceive("fetch")->withArgs([3, 3])->andReturn([$records[3]]);
+        $store->shouldReceive("fetch")->withArgs([6, 3])->andReturn([]);
+        $store->shouldReceive("save");
```

Comment:
use the ordered method call for this to make sure things are called in the right order also use the call count verification methods

```
$store->shouldReceive("fetch")->once()->ordered()->with(0, 3)->andReturn(
            [
                $records[0],
                $records[1],
                $records[2],
            ]);
        $store->shouldReceive("save")->once()->ordered()->with(new EqualsMatcher(
[
                $rotatedRecords[0],
                $rotatedRecords[1],
                $rotatedRecords[2],
            ]
        ));
        $store->shouldReceive("fetch")->once()->ordered()->with(3, 3)->andReturn([$records[3]]);
        $store->shouldReceive("save")->once()->ordered()->with(new EqualsMatcher(
        [
                $rotatedRecords[3]
            ]
        ));
        $store->shouldReceive("fetch")->once()->ordered()->with(6, 3)->andReturn([]);